$NetBSD: patch-Makefile,v 1.2 2025/10/03 13:40:04 dkazankov Exp $

Make BUILD_MODE externally settable
Add ALL_LIBRARY_TYPES to build several libraries
Add options for gprbuild and gprinstall
Fix GPRINSTALL parameters
Remove test specs download

--- Makefile.orig	2024-05-04 22:43:19.000000000 +0300
+++ Makefile
@@ -5,7 +5,8 @@
 #
 
 # Build mode (dev, prod or coverage)
-BUILD_MODE=dev
+BUILD_MODE?=dev
+ALL_LIBRARY_TYPES?=static static-pic relocatable
 
 GPRBUILD_FLAGS = -p -j0 -XBUILD_MODE=$(BUILD_MODE)
 PREFIX                 ?= /usr
@@ -26,17 +27,31 @@
 .PHONY: spellcheck check
 
 all:
-	gprbuild $(GPRBUILD_FLAGS) gnat/markdown.gpr -cargs $(ADAFLAGS)
+	for library_type in $(ALL_LIBRARY_TYPES) ; do \
+	    gprbuild \
+		$(GPRBUILD_FLAGS) -XLIBRARY_TYPE=$$library_type \
+		gnat/markdown.gpr \
+		-cargs $(ADAFLAGS) -gargs $(GPRBUILD_OPTIONS) ; \
+	done;
 
 install:
-	gprinstall $(GPRINSTALL_FLAGS) -p -P gnat/markdown.gpr
+	for library_type in $(ALL_LIBRARY_TYPES) ; do \
+	    gprinstall \
+		--prefix=$(DESTDIR)$(PREFIX) \
+		-XLIBRARY_TYPE=$$library_type \
+		--build-name=$$library_type \
+		--build-var=LIBRARY_TYPE \
+		-p -P \
+		gnat/markdown.gpr \
+		$(GPRINSTALL_OPTIONS) ; \
+	done;
 
 build_tests: all
-	gprbuild $(GPRBUILD_FLAGS) -aP gnat -P gnat/tests/commonmark_tests.gpr
+	gprbuild $(GPRBUILD_FLAGS) -aP gnat -P gnat/tests/commonmark_tests.gpr $(GPRBUILD_OPTIONS)
 
 check: build_tests check_markdown
 
-check_markdown: commonmark-spec
+check_markdown:
 	cd commonmark-spec; python3 test/spec_tests.py --program ../.objs/static/tests/commonmark_tests |\
 	  grep -E "^Example|^[0-9]+.passed" |\
 	  tee markdown_tests_result | tail
